src/builders.ts

```diff
@@ -160,18 +156,71 @@ export class BomBuilder {
       report: new ThrowReport(),
       cacheOptions: { skipIntegrityCheck: true }
     }
-    const gatherLicenseTexts = this.gatherLicenseTexts
-    return async function (pkg: Package): Promise<PackageInfo> {
+    return async function (pkg: Package): Promise<any> {
       const { packageFs, prefixPath, releaseFs } = await fetcher.fetch(pkg, fetcherOptions)
       try {
         const manifestPath = ppath.join(prefixPath, 'package.json')
-        const packageInfo: PackageInfo = {
-          manifest: JSON.parse(await packageFs.readFilePromise(manifestPath, 'utf8'))
+        return JSON.parse(await packageFs.readFilePromise(manifestPath, 'utf8'))
+      } finally {
+        if (releaseFs !== undefined) {
+          releaseFs()
         }
-        if (gatherLicenseTexts) {
-          packageInfo.licenseEvidence = makeLicenseEvidence(prefixPath, packageFs)
+      }
+    }
+  }
+
+  readonly #LICENSE_FILENAME_PATTERN = /^(?:UN)?LICEN[CS]E|.\.LICEN[CS]E$|^NOTICE$/i
+
+  private async makeLicenseEvidenceFetcher (project: Project): Promise<LicenseEvidenceFetcher> {
+    const fetcher = project.configuration.makeFetcher()
+    const fetcherOptions: FetchOptions = {
+      project,
+      fetcher,
+      cache: await Cache.find(project.configuration),
+      checksums: project.storedChecksums,
+      report: new ThrowReport(),
+      cacheOptions: { skipIntegrityCheck: true }
+    }
+    const LICENSE_FILENAME_PATTERN = this.#LICENSE_FILENAME_PATTERN
+    const logger = this.console
+    return async function * (pkg: Package): AsyncGenerator<License> {
+      const { packageFs, prefixPath, releaseFs } = await fetcher.fetch(pkg, fetcherOptions)
+      try {
+        let files
+        try {
+          // option `withFileTypes:true` is not supported and casues crashes
+          files = packageFs.readdirSync(prefixPath)
+        } catch (e) {
```

Comment:
This catch should not be here as it hides errors. The warning message is not enough.

I'd assume users of cyclonedx-node-yarn would call it as part of a build pipeline and expect a non zero exit code if anything fails that affects the produced SBOM. It's better to explode than to log a human-friendly message.

Reply:
I copied this over from webpack implementation, without thinking too much.
With webpack, the SBOM is a side-artifact that is generated during build time, and it is designed to not break build processes.

the yarn plugin is called explicitly, and - like you said - it is expected to explode on errors.
I will change this accordingly.